yorkie/src/main/kotlin/dev/yorkie/document/Document.kt

```diff
@@ -91,14 +98,82 @@ public class Document(public val key: Key) {
                 return@async true
             }
             val change = context.getChange()
-            change.execute(root)
+            val operationInfos = change.execute(root)
             localChanges += change
             changeID = change.id
-            eventStream.emit(change.asLocal())
+            val changeInfos = listOf(change.toChangeInfo(operationInfos))
+            eventStream.emit(Event.LocalChange(changeInfos))
             true
         }
     }
 
+    /**
+     * Subscribes to events on the document with the specific [targetPath].
+     */
+    public fun events(targetPath: String): SharedFlow<Event> {
+        return targetEventStreams.getOrElse(targetPath) {
```

Comment:
I’m not sure if this is the common way of dealing shared flow, such as returning shared flow from a function or storing them in a map. 
I’ll refactor it if it seems awkward or if there could be potential issues.

Reply:
We can use `getOrPut` to skip manually setting the new value at the end, but I don't think we need `targetEventStreams`, as sharing a SharedFlow instance for each path seems unnecessary.
Just returning a filtered Flow without eagerly sharing should be enough.